main.nf

```diff
@@ -1236,28 +1244,36 @@ if (params.binning_map_mode == 'all'){
 
 process bowtie2 {
     tag "$assembler-$name"
+    publishDir "${params.outdir}/Assembly/${assembler}/${name}_QC", mode: params.publish_dir_mode,
+        saveAs: {filename -> filename.indexOf(".bowtie2-log") > 0 ? filename : null}
 
     input:
     set val(assembler), val(name), val(group), file(assembly), val(sampleToMap), val(sampleGroup), file(reads) from ch_bowtie2_input
 
     output:
     set val(assembler), val(name), file("${assembler}-${name}-${sampleToMap}.bam"), file("${assembler}-${name}-${sampleToMap}.bam.bai") into ch_assembly_mapping_for_metabat
+    set val(assembler), val(name), val(sampleToMap), file("*.bowtie2-log") into ch_assembly_mapping_stats
 
     when:
     !params.skip_binning
 
     script:
-    def name = "${assembler}-${name}-${sampleToMap}"
+    def NAME = "${assembler}-${name}-${sampleToMap}"
     def input = params.single_end ? "-U \"${reads}\"" :  "-1 \"${reads[0]}\" -2 \"${reads[1]}\""
         """
         bowtie2-build --threads "${task.cpus}" "${assembly}" ref
-        bowtie2 -p "${task.cpus}" -x ref $input | \
+        bowtie2 -p "${task.cpus}" -x ref $input 2>"${NAME}.bowtie2-log" | \
             samtools view -@ "${task.cpus}" -bS | \
-            samtools sort -@ "${task.cpus}" -o "${name}.bam"
-        samtools index "${name}.bam"
+            samtools sort -@ "${task.cpus}" -o "${NAME}.bam"
+        samtools index "${NAME}.bam"
+        if [ ${NAME} = "${assembler}-${name}-${name}" ] ; then
+            mv "${NAME}.bowtie2-log" "${assembler}-${name}.bowtie2-log"
+        fi
         """
 }
 
+ch_assembly_mapping_stats_for_multiqc = ch_assembly_mapping_stats.map { assembler, name, sampleToMap ,file -> if (name == sampleToMap) {return [ file ]} }
```

Comment:
So this would only report mappings, if no co-assembly is done and, additionally, only for the sample the assembly is created from. 

In case of `--binning_map_mode  all` this would mean that only a subset of the used mappings is reported, right?  This seems a bit confusing to me.

Reply:
All correct, I have written this in the opening post.

> This is nicely integrated into the MultiQC report when assembling individual samples but not when assembling groups. Because it is here impossible to match assembly to sample and would create a huge mess.

As it is with this setting, the MultiQC report shows an additional column with "% Aligned" for each assembly, how I would expect. I do not care if other samples map at this point.
If this isnt done, the MultiQC report shows (number of sample)² rows with only entries in "% Aligned", completely unacceptable in my opinion.

In case of --binning_map_mode all, this isnt shown at all because it makes no sense at all to show the mapping of just one sample, but I strongly dislike adding almost empty row to the report. I think we should eventually add this to the summary table "quast_and_busco_summary.tsv"

Reply:
ok